test/test_build.py

```diff
@@ -66,9 +73,29 @@ def image_type_fixture(shared_tmpdir, build_container, request, force_aws_upload
     """
     Build an image inside the passed build_container and return an
     ImageBuildResult with the resulting image path and user/password
+    In the case an image is being built from a local container, the
+    function will build the required local container for the test.
     """
-    with build_images(shared_tmpdir, build_container, request, force_aws_upload) as build_results:
-        yield build_results[0]
+    container_ref, images, target_arch, local = parse_request_params(request)
+
+    if not local:
+        with build_images(shared_tmpdir, build_container, request, force_aws_upload) as build_results:
```

Comment:
I think it's possible to deduplicate these two lines by moving them after the if-else block.